lbrynet/database/migrator/migrate5to6.py

```diff
@@ -0,0 +1,164 @@
+import sqlite3
+import os
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.StreamHandler())
+log.setLevel(logging.INFO)
+
+
+def do_migration(db_dir):
+    log.info("Doing the migration")
+    db_refactor(db_dir)
+    log.info("Migration succeeded")
+
+
+def db_refactor(db_dir):
+    new_db_path = os.path.join(db_dir, "lbrynet.sqlite")
+
+    name_metadata_path = os.path.join(db_dir, "blockchainname.db")
+    lbryfile_info_db_path = os.path.join(db_dir, 'lbryfile_info.db')
+    blobs_db_path = os.path.join(db_dir, 'blobs.db')
+
+    name_metadata_db = sqlite3.connect(name_metadata_path)
+    lbryfile_db = sqlite3.connect(lbryfile_info_db_path)
+    new_db = sqlite3.connect(new_db_path)
+    blobs_db = sqlite3.connect(blobs_db_path)
+
+    name_metadata_cursor = name_metadata_db.cursor()
+    lbryfile_cursor = lbryfile_db.cursor()
+    new_db_cursor = new_db.cursor()
+    blobs_db_cursor = blobs_db.cursor()
+
+    new_db.executescript("""
+            pragma foreign_keys = on;
+            pragma journal_mode=WAL
+    
+            create table blob (
+                blob_hash char(96) primary key not null,
+                blob_length integer not null,
+                next_announce_time integer not null,
+                should_announce integer not null default 0
+            );
+            
+            create table stream (
+                stream_hash char(96) not null primary key,
+                sd_hash char(96) not null,
+                stream_key text not null,
+                stream_name text not null,
+                suggested_filename text not null,
+                foreign key(sd_hash) references blob(blob_hash)
+            );
+            
+            create table stream_blob (
+                stream_hash char(96) not null,
+                blob_hash char(96),
```

Comment:
my note disappeared somehow. `blob_hash` should be not null. a stream_blob's purpose is to connect a stream and a blob. if there's no blob, there's no stream_blob

Reply:
@jackrobison this still applies

Reply:
this is to handle stream terminators